icicle_v3/backend/cpu/src/curve/cpu_msm.hpp

```diff
@@ -0,0 +1,667 @@
+#pragma once
+
+#include <thread>
+#include <atomic>
+#include <string>
+#include <iostream>
+#include <fstream>
+
+#include "icicle/errors.h"
+#include "icicle/runtime.h"
+#include "icicle/config_extension.h"
+#include "icicle/curves/projective.h"
+#include "icicle/curves/curve_config.h"
+#include "icicle/msm.h"
+#include "tasks_manager.h"
+
+using namespace icicle;
+
+#ifdef DUMMY_TYPES // for testing
+using A = DummyP;
+using P = DummyP;
+using scalar_t = DummyScalar;
+#else
+using A = curve_config::affine_t;
+using P = curve_config::projective_t;
+using scalar_t = curve_config::scalar_t;
+#endif
+
+/**
+ * @class EcAddTask
+ * @brief class deriving `TaskBase` to allow use of tasks_manager.h for MSM
+ * (The task to be executed is an elliptic curve addition).
+ */
+template <typename A, typename P>
+class EcAddTask : public TaskBase
+{
+public:
+  /**
+   * @brief constructor for the task ensuring Ps are zeros and other fields are init to invalid values (to avoid
+   * falsely handling a result at the start). The execution adds Ps m_p1,m_p2 and stores the result in m_p1.
+   */
+  EcAddTask() : TaskBase(), m_p1(P::zero()), m_p2(P::zero()), m_return_idx(-1) {}
+
+  /**
+   * @brief Function to be executed by the tasks manager. It can be configured according to the value odf
+   */
+  void execute()
+  {
+    switch (m_p2_config) {
```

Comment:
(1) use better names for the enum values. what is PER?
(2) ADD_P1_AND_AFFINE_P1 adds p1 and p2?
(3) I don't get the name  'm_p2_config'

Reply:
1. typo
2. adds p1 and affine p2
3. it's to config which p2 will be added

Reply:
about (2), I meant the typo